README.rst

```diff
@@ -30,7 +30,94 @@ dashboard, please contact your Optimizely account executive.
 Using the SDK
 ~~~~~~~~~~~~~
 
-See the Optimizely `Full Stack documentation`_ to learn how to
+You can initialize the Optimizely instance in three ways: with a datafile, by providing an `sdk_key`, or by providing a Config Manager. Each method is described below. 
+
+1. Initialize Optimizely with a datafile. This datafile will be used as
+   ProjectConfig throughout the life of Optimizely instance.
+   ::
+
+      optimizely.Optimizely(
+        datafile
+      )
+
+2. Initialize Optimizely by providing an 'sdk_key'. This will initialize
+   a PollingConfigManager that makes an HTTP GET request to the URL (formed 
+    using your provided `sdk key` and the default datafile CDN url
+   template) to asynchronously download the project datafile at regular
+   intervals and update ProjectConfig when a new datafile is recieved. A
+   hard-coded datafile can also be provided along with the `sdk_key` that
+   will be used initially before any update.
+   ::
+
+      optimizely.Optimizely(
+        datafile=None,
+        sdk_key='put_your_sdk_key_here'
+      )
+
+3. Initialize Optimizely by providing a Config Manager that implements a
```

Comment:
Instead, link to `BaseConfigManager` and ask for user to implement `BaseConfigManager`.